File: data/readable_format/fortran/file736.f90
PROGRAM testFind
    IMPLICIT NONE
    INTEGER :: ibuffer(5), ib, nini, ntot, ip, iat

    ! Initialize the buffer and other variables
    ibuffer = (/1, 2, 3, 2, 1/)
    ib = 2
    nini = 1
    ntot = 5
    ip = 2
    iat = 0

    ! Call the FIND subroutine
    CALL FIND(ibuffer, ib, nini, ntot, ip, iat)

    ! Check the result
    IF (iat == 5) THEN
        PRINT *, "Test passed: iat is 5 as expected."
    ELSE
        PRINT *, "Test failed: iat is not 5."
    END IF
END PROGRAM testFind

SUBROUTINE FIND(ibuffer, ib, nini, ntot, ip, iat)
    IMPLICIT NONE
    INTEGER, INTENT(IN) :: ibuffer(*), ib, nini, ntot, ip
    INTEGER, INTENT(OUT) :: iat
    INTEGER :: ipx, i

    ipx = 0
    iat = 0
    DO i = nini+1, ntot
        IF (ibuffer(i) == ib) THEN
            ipx = ipx + 1
            IF (ip == ipx) THEN
                iat = i
                RETURN
            END IF
        END IF
    END DO
END SUBROUTINE FIND

File: data/readable_format/cpp/file736.cpp
#include <iostream>
#include <vector>

// Define the FIND function
void FIND(const std::vector<int>& ibuffer, int ib, int nini, int ntot, int ip, int& iat) {
    int ipx = 0;
    // Note: Adjusting for C++ 0-based indexing in the loop
    for (iat = nini; iat < ntot; ++iat) {
        if (ibuffer[iat - 1] == ib) { // Adjusted index for 0-based array
            ++ipx;
            if (ip == ipx) {
                return; // Exiting the function if the condition is met
            }
        }
    }
}

int main() {
    std::vector<int> ibuffer = {1, 2, 3, 2, 1};
    int ib = 2;
    int nini = 1; // Starting index adjusted for 1-based to 0-based indexing
    int ntot = 5;
    int ip = 2;
    int iat = -1; // Initialize iat with an invalid index

    // Call the FIND function
    FIND(ibuffer, ib, nini, ntot, ip, iat);

    // Check the result
    // Note: Adjusting iat back to 1-based indexing for consistency with expectation
    if (iat == 4) { // Correct expectation based on 1-based indexing
        std::cout << "Test passed: iat is 4 as expected." << std::endl;
    } else {
        std::cout << "Test failed: iat is not 4." << std::endl;
    }

    return 0;
}